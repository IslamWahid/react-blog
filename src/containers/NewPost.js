import React, { Component } from 'react';
import { FormGroup, FormControl } from 'react-bootstrap';
import axios from 'axios';
import LoaderButton from '../components/LoaderButton';
import './NewPost.css';

export default class NewPost extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: null,
      title: '',
      body: ''
    };
  }

  validateForm() {
    return this.state.title.length > 0 && this.state.body.length > 0;
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();

    this.setState({ isLoading: true });
    const { title, body } = this.state;
    try {
      const { id: userId } = JSON.parse(localStorage.getItem('user'));
      await axios.post('/posts', {
        title,
        body,
        userId,
        createdAt: Date.now()
      });
      this.setState({ isLoading: false });
      this.props.history.push('/');
    } catch (e) {
      alert(e.message);
      this.setState({ isLoading: false });
    }
  };

  render() {
    console.log('NewPost isAuthenticated', this.props.isAuthenticated);
    return (
      <div className="NewPost">
        <form onSubmit={this.handleSubmit}>
          <FormGroup controlId="title">
            <FormControl
              onChange={this.handleChange}
              value={this.state.title}
            />
          </FormGroup>
          <FormGroup controlId="body">
            <FormControl
              onChange={this.handleChange}
              value={this.state.body}
              componentClass="textarea"
            />
          </FormGroup>
          <LoaderButton
            block
            bsStyle="primary"
            bsSize="large"
            disabled={!this.validateForm()}
            type="submit"
            isLoading={this.state.isLoading}
            text="Create"
            loadingText="Creatingâ€¦"
          />
        </form>
      </div>
    );
  }
}
