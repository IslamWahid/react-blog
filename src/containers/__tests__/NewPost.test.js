import React from 'react';
import NewPost from '../NewPost';
import { shallow } from 'enzyme';

describe('NewPost Component', () => {
  let newPost;

  beforeEach(() => {
    newPost = shallow(<NewPost />);
  });
  it('should match snapshot', () => {
    expect(newPost).toMatchSnapshot();
  });
});
