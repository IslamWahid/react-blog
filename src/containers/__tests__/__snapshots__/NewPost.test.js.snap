// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPost Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewPost />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <FormGroup
          bsClass="form-group"
          controlId="title"
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            value=""
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="body"
        >
          <FormControl
            bsClass="form-control"
            componentClass="textarea"
            onChange={[Function]}
            value=""
          />
        </FormGroup>
        <_default
          block={true}
          bsSize="large"
          bsStyle="primary"
          disabled={true}
          isLoading={null}
          loadingText="Creating…"
          text="Create"
          type="submit"
        />
      </form>,
      "className": "NewPost",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
            controlId="title"
          >
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              value=""
            />
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            controlId="body"
          >
            <FormControl
              bsClass="form-control"
              componentClass="textarea"
              onChange={[Function]}
              value=""
            />
          </FormGroup>,
          <_default
            block={true}
            bsSize="large"
            bsStyle="primary"
            disabled={true}
            isLoading={null}
            loadingText="Creating…"
            text="Create"
            type="submit"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              value=""
            />,
            "controlId": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "input",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              componentClass="textarea"
              onChange={[Function]}
              value=""
            />,
            "controlId": "body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "textarea",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": true,
            "bsSize": "large",
            "bsStyle": "primary",
            "disabled": true,
            "isLoading": null,
            "loadingText": "Creating…",
            "text": "Create",
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <FormGroup
            bsClass="form-group"
            controlId="title"
          >
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              value=""
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="body"
          >
            <FormControl
              bsClass="form-control"
              componentClass="textarea"
              onChange={[Function]}
              value=""
            />
          </FormGroup>
          <_default
            block={true}
            bsSize="large"
            bsStyle="primary"
            disabled={true}
            isLoading={null}
            loadingText="Creating…"
            text="Create"
            type="submit"
          />
        </form>,
        "className": "NewPost",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              controlId="title"
            >
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                value=""
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
              controlId="body"
            >
              <FormControl
                bsClass="form-control"
                componentClass="textarea"
                onChange={[Function]}
                value=""
              />
            </FormGroup>,
            <_default
              block={true}
              bsSize="large"
              bsStyle="primary"
              disabled={true}
              isLoading={null}
              loadingText="Creating…"
              text="Create"
              type="submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                value=""
              />,
              "controlId": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": <FormControl
                bsClass="form-control"
                componentClass="textarea"
                onChange={[Function]}
                value=""
              />,
              "controlId": "body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "textarea",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "block": true,
              "bsSize": "large",
              "bsStyle": "primary",
              "disabled": true,
              "isLoading": null,
              "loadingText": "Creating…",
              "text": "Create",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
